tasks/query_resources.json

```diff
@@ -0,0 +1,22 @@
+{
+  "description": "Helper to retrieve resources from a Target(for internal use only)",
+  "input_method": "stdin",
+  "private": true,
+  "supports_noop": true,
+  "implementations": [
+    { "name": "query_resources.rb" },
+    { "name": "query_resources.rb", "remote": true}
+  ],
+  "parameters": {
+    "plugins": {
+      "description": "Custom facts bundle to use when applying resources",
+      "type": "String",
+      "sensitive": true
+    },
+    "resources": {
+      "description": " A resource type or instance, or an array of such",
+      "type": "Variant[String, Type[Resource], Array[Variant[String, Type[Resource]]]]"
```

Comment:
Hmm, why not simplify this to `Array[Variant[String, Type[Resource]]]`? I also don't think the existing code would work with a non-array argument b/c of the `flat_map` call.

Reply:
ah, good call. yeah looks like the plan function ensures there is a flatt array of that type. Will update thanks. https://github.com/puppetlabs/bolt/blob/a9ae282d31d993b27befa8c4f22ffc7d90a95037/bolt-modules/boltlib/lib/puppet/functions/get_resources.rb#L22

Reply:
It's actually just `Array[String]`. The function stringifies them all. https://github.com/puppetlabs/bolt/blob/a9ae282d31d993b27befa8c4f22ffc7d90a95037/bolt-modules/boltlib/lib/puppet/functions/get_resources.rb#L53

`Type[Resource]` isn't `Data` so you can't pass it as a JSON task parameter. I actually thought there would be an error somewhere if the specified type wasn't a subset of `Data` but maybe not.

Reply:
aww yeah, I will get a pr up to fix that. My bad